src/main/java/com/treasuredata/client/TDHttpClient.java

```diff
@@ -450,38 +464,48 @@ private TDClientException handleHttpResponseError(String apiRequestPath, int cod
                     String conflictsWith = errorResponse.isPresent() ? parseConflictsWith(errorResponse.get()) : null;
                     throw new TDClientHttpConflictException(errorMessage, conflictsWith);
                 case HttpStatus.PROXY_AUTHENTICATION_REQUIRED_407:
-                    throw new TDClientHttpException(PROXY_AUTHENTICATION_FAILURE, errorMessage, code);
+                    throw new TDClientHttpException(PROXY_AUTHENTICATION_FAILURE, errorMessage, code, retryAfter);
```

Comment:
Oops. I was confused with TDClientHttpException and other exception classes in above `case` statements.